core-processor/src/ext.rs

```diff
@@ -611,11 +706,22 @@ impl Ext {
         }
     }
 
-    fn charge_sending_fee(&mut self, delay: u32) -> Result<(), ChargeError> {
+    fn charge_sending_fee(
+        &mut self,
+        changes: &mut ExtChanges,
+        delay: u32,
+    ) -> Result<(), ChargeError> {
         if delay == 0 {
-            self.charge_gas_if_enough(self.context.message_context.settings().sending_fee)
+            changes.charge_gas_if_enough(self.context.message_context.settings().sending_fee)
```

Comment:
what if we impl charge_sending_fee directly on `ExtChanges` (btw I'd consider renaming into `ExtOverlay`)?